testapi.pm

```diff
@@ -1549,6 +1571,72 @@ sub mouse_hide(;$) {
+the mouse to another location where the button is released. You can set the C<$startpoint> 
+and C<$endpoint> by passing the name of the needle tag, i.e. the mouse drag happens between
+the two needle areas. Alternatively, you can set all the coordinates explicitly with C<$startx>,
+C<$starty>, C<$endx>, and C<$endy>. You can also set one point using a needle and another one
+using coordinates.  If both the coordinates and the needle are provided, the coordinates 
+will be used to set up the locations and the needle location will be overridden.
+
+=cut
+
+sub mouse_drag {
+    my %args = @_;
+    my $startx =
+      my $starty =
+      my $endx =
+      my $endy = 0;
+    # If full coordinates are provided, work with them as a priority,
+    if (defined $args{startx} and defined $args{starty}) {
+        $startx = $args{startx};
+        $starty = $args{starty};
+    }
+    # If the coordinates were not complete, use the needle as a fallback solution.
+    elsif (defined $args{startpoint}) {
+        my $startmatch = $args{startpoint};
+        # Check that the needle exists.
+        my $start_matched_needle = assert_screen($startmatch, $args{timeout});
+        # Calculate the click point from the area defined by the needle (take the center of it)
+        ($startx, $starty) = calculate_clickpoint($start_matched_needle);
+    }
+    # If neither coordinates nor a needle is provided, report an error and quit.
+    else {
+        die "The starting point of the drag was not correctly provided. Either provide the 'startx' and 'starty' coordinates, or a needle marking the starting point.";
+    }
+
+    # Repeat the same for endpoint coordinates or needles.
+    if (defined $args{endx} and defined $args{endy}) {
+        $endx = $args{endx};
+        $endy = $args{endy};
+    }
+    elsif (defined $args{endpoint}) {
+        my $endmatch           = $args{endpoint};
+        my $end_matched_needle = assert_screen($endmatch, $args{timeout});
+        ($endx, $endy) = calculate_clickpoint($end_matched_needle);
+    }
+    else {
+        die "The ending point of the drag was not correctly provided. Either provide the 'endx' and 'endy' coordinates, or a needle marking the starting point.";
+    }
+    # Get the button variable. If no button has been provided, assume the "left" button.
+    my $button = $args{button} //= "left";
+    bmwqemu::log_call("mouse dragged", button => $button);
+
+    # Now, perform the actual mouse drag. Navigate to the startpoint location,
+    # press and hold the mouse button, then navigate to the endpoint location
+    # and release the mouse button.
+    mouse_set($startx, $starty);
+    query_isotovideo('backend_mouse_button', {button => $button, bstate => 1});
+    mouse_set($endx, $endy);
+    query_isotovideo('backend_mouse_button', {button => $button, bstate => 0});
+
+    # Sleep for a while to leave a trace on the SUT video.
+    sleep("5");
```

Comment:
Sleeping isn't nice so I would at least make the time configurable. And is it really just for the video? Because then it is questionable whether leaving a trace in the video justifies slowing down the test by 5 seconds. (Test are already slow enough.)

Reply:
Ok, the sleeping time is totally unnecessary here, cause anybody could put the sleep in their tests, if they wanted. I will delete it.